File: ContestOfSwift/ContestOfSwift/South/LanguageExtensions.swift
//
//  LanguageExtensions.swift
//  ContestOfSwift
//
//  Created by Sergio Daniel on 5/5/24.
//

import Foundation

// TO TEST
public func when<T>(_ rules: [(Bool, T)], fallbackingWith fallback: T) -> T {
    rules.first { $0.0 }?.1 ?? fallback
}

File: ContestOfSharp/ContestOfSharp/South/LanguageExtensions.cs
namespace ContestOfSharp.South;

public static class LanguageExtensions
{
    public static T When<T>((bool condition, T value)[] rules, T fallback)
        => rules
            .Where(r => r.condition)
            .Select(r => r.value)
            .FirstOrDefault() ?? fallback;
}